prime-router/src/main/kotlin/cli/TestReportStream.kt

```diff
@@ -1356,4 +1396,145 @@ class InternationalContent : CoolTest() {
             return bad("***intcontent Test FAILED***: There was an error fetching data from the database.")
         }
     }
+}
+
+class SantaClaus : CoolTest() {
+
+    override val name = "santaclaus"
+    override val description = "Creates fake data as if from a sender and tries to send it to every state and territory"
+    override val status = TestStatus.GOODSTUFF
+
+    override fun run(environment: ReportStreamEnv, options: CoolTestOptions): Boolean {
+
+        var passed = true
+
+        if (options.env == "prod") {
+            return createBad("This test can only be run on staging or locally")
+        }
+
+        var sendersToTestWith = settings.senders.filter {
+            it.organizationName in listOf("simple_report", "waters", "strac", "safehealth")
+        }
+
+        options.sender?.let {
+
+            // validates sender existence
+            val sender = settings.findSender(it) ?: return createBad("The sender indicated doesn't exists '$it'")
+
+            // replace the list of senders to test
+            // with the indicated by parameter
+            sendersToTestWith = listOf(sender)
+        }
+
+        val states = metadata.findLookupTable("fips-county")?.getDistinctValuesInColumn("State")?.toList() ?: listOf()
+
+        sendersToTestWith.forEach { sender ->
+
+            states.forEach { state ->
+
+                ugly("Starting $name Test: send ${sender.fullName} data to $state")
+
+                val report = FileUtilities.createFakeReport(
+                    metadata,
+                    sender,
+                    1,
```

Comment:
Good news:  You do NOT need to create one file per state.   Remember our discussion about creating a file with 57 rows (or 56 or whatever)?   If you do that, and don't specify any target-states, the Fake report generator will create one row for each state/territory.

Then you don't need writeReportToByteArray either, because you are only creating one file!   (Or 4 files, in the case of the default), which means you can remove a lot of the changes you made to FileUtilities.kt and other places.

You will have to change your query at the end to check for 1 row for each state/territory.    And best of all, this will run _much_ faster!

Reply:
updated and being done as requested